src/Ushahidi/Modules/V5/Actions/Post/Handlers/AbstractPostQueryHandler.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace Ushahidi\Modules\V5\Actions\Post\Handlers;
+
+use App\Bus\Action;
+use App\Bus\Query\Query;
+use Ushahidi\Modules\V5\Actions\V5QueryHandler;
+use Illuminate\Support\Facades\Auth;
+use Ushahidi\Modules\V5\Models\RolePermission;
+
+abstract class AbstractPostQueryHandler extends V5QueryHandler
+{
+    protected function userHasManagePostPermissions()
+    {
+        $user = Auth::user();
+        if (!$user || !$user->id) {
+            return false;
+        }
+        if ($user->role === "admin") {
+            return true;
+        }
+        $permissions =
+            RolePermission::select("permission")->where('role', '=', $user->role)->get()->pluck('permission');
+        if (in_array("Manage Posts", $permissions->toArray())) {
+            return true;
+        }
+        return false;
```

Comment:
This piece of code already exists in Ushahidi\Core\Tool\Acl class so no need for code duplication

Reply:
See here https://github.com/ushahidi/platform/blob/88c5381da67bf95bd28dc9f637282308a2357bcf/src/Ushahidi/Core/Tool/Acl.php#L36-L56

So the best way would be to inject the Acl class via the constructor into the handler, Laravel Container handles the resolution, and we can the do permission checks

`$this->accessControl->hasPermission($user, 'Manage Post');`